### ISSUE TITLE
summary.add_activation_summary not working?

### ISSUE THREAD

**Issue by byangderek** [ppwwyyxx/tensorpack#206]
In [mnist-convnet](https://github.com/ppwwyyxx/tensorpack/blob/master/examples/mnist-convnet.py) example, if I want to use summary.add_activation_summary, I simply add
`summary.add_activation_summary(logits)`
at [line96](https://github.com/ppwwyyxx/tensorpack/blob/master/examples/mnist-convnet.py#L96),
but it raised an error saying that the input placeholders are empty. 
Seems like summary.add_activation_summary should be implemented inside the inference_runner but is not?

**Comment by ppwwyyxx** [comment]
Often a summary tensor doesn't require input tensors to evaluate (e.g. summary of variables). But in this case the summary requires input tensors to evaluate. Then you'll face two options:
1. evaluate on a new input data taking from the dataflow
2. evaluate together with training (i.e. reuse a datapoint to evaluate the summary)
In most applications they don't make much difference, but in certain cases the difference is important.

To use 1., you can simply replace SimpleTrainer with QueueInputTrainer. Because of how tensorflow queue works, the queue will automatically produce the next data when it is needed.
To use 2., you can explicitly tell the trainer that the summary callback should be run together with training. This is done by overriding the [default call back list](http://tensorpack.readthedocs.io/en/latest/modules/train.html#tensorpack.train.TrainConfig.__init__):
```python
        extra_callbacks=[
            MovingAverageSummary(),
            ProgressBar(),
            MergeAllSummaries(run_alone=True)
        ],
```

**Comment by ppwwyyxx** [comment]
A bug was introduced lately. It always tries to evaluate the summary tensor in a separate sess.run call, which leads to error because the call may depend on feed_dict.
